libraries/AP_Math/AP_Math.cpp

```diff
@@ -82,13 +82,28 @@ template float safe_sqrt<short>(const short v);
 template float safe_sqrt<float>(const float v);
 template float safe_sqrt<double>(const double v);
 
+/*
+  replacement for std::swap() needed for STM32
+ */
+static void swap_float(float &f1, float &f2)
+{
+    float tmp = f1;
+    f1 = f2;
+    f2 = tmp;
+}
+
 /*
  * linear interpolation based on a variable in a range
  */
 float linear_interpolate(float low_output, float high_output,
                          float var_value,
                          float var_low, float var_high)
 {
+    if (var_low > var_high) {
```

Comment:
Why are we getting into this situation? Given how much we use linear interpolate it seems a bit worrisome to

A: hit the situation
B: it seems like an extra cost we shouldn't be paying...

Reply:
we can use user settable parameters as the two endpoints we are interpolating between, and we want to ensure we get sane behaviour if the two are in an unexpected order. For example, its valid for a user to setup landing approach speed to be higher than cruise speed. 
The other key thing is API robustness, where the programmer needs to think carefully about increasing vs decreasing variable. 
even though I originally added linear_interpolate() I had forgotten it wasn't symmetric and made a mistake in the ordering. I don't want others to make the same mistake